stress: failed test in cockroach/gossip/gossip.test: TestGossipOrphanedStallDetection

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/9eddee873088626f33857a6f466a5c11f8cf3ea1

Stress build found a failed test:

```
=== RUN   TestGossipOrphanedStallDetection
W160831 05:25:52.573851 gossip/gossip.go:1037  no resolvers found; use --join to specify a connected node
W160831 05:25:52.575176 gossip/gossip.go:1037  no resolvers found; use --join to specify a connected node
I160831 05:25:52.577222 gossip/client.go:123  node 2: started gossip client to 127.0.0.1:36483
I160831 05:25:52.577694 stopper.go:408  quiesceing; tasks left:
2      gossip/infostore.go:290
I160831 05:25:52.577834 http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160831 05:25:52.577870 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:36483->127.0.0.1:48522: use of closed network connection
I160831 05:25:52.578484 /go/src/google.golang.org/grpc/clientconn.go:667  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:36483: getsockopt: connection refused"; Reconnecting to {"127.0.0.1:36483" <nil>}
W160831 05:25:52.578554 gossip/gossip.go:1037  no resolvers found; use --join to specify a connected node
I160831 05:26:37.663219 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:36483->127.0.0.1:49166: use of closed network connection
I160831 05:26:37.663236 http2_client.go:1053  transport: http2Client.notifyError got notified that the client transport was broken EOF.
--- FAIL: TestGossipOrphanedStallDetection (45.09s)
	testing.go:117: gossip/gossip_test.go:321, condition failed to evaluate within 45s: 2 not yet connected
```

Run Details:

```
29 runs so far, 0 failures, over 5s
72 runs so far, 0 failures, over 10s
118 runs so far, 0 failures, over 15s
158 runs so far, 0 failures, over 20s
199 runs so far, 0 failures, over 25s
239 runs so far, 0 failures, over 30s
285 runs so far, 0 failures, over 35s
323 runs so far, 0 failures, over 40s
364 runs so far, 0 failures, over 45s
408 runs so far, 0 failures, over 50s
450 runs so far, 0 failures, over 55s
492 runs so far, 0 failures, over 1m0s
534 runs so far, 0 failures, over 1m5s
576 runs so far, 0 failures, over 1m10s
617 runs so far, 0 failures, over 1m15s
659 runs so far, 0 failures, over 1m20s
704 runs so far, 0 failures, over 1m25s
744 runs so far, 0 failures, over 1m30s
787 runs so far, 0 failures, over 1m35s
829 runs so far, 0 failures, over 1m40s
870 runs so far, 0 failures, over 1m45s
915 runs so far, 0 failures, over 1m50s
961 runs so far, 0 failures, over 1m55s
1001 runs so far, 0 failures, over 2m0s
1043 runs so far, 0 failures, over 2m5s
1082 runs so far, 0 failures, over 2m10s
1125 runs so far, 0 failures, over 2m15s
1167 runs so far, 0 failures, over 2m20s
1209 runs so far, 0 failures, over 2m25s
1252 runs so far, 0 failures, over 2m30s
1297 runs so far, 0 failures, over 2m35s
1341 runs so far, 0 failures, over 2m40s
1378 runs so far, 0 failures, over 2m45s
1417 runs so far, 0 failures, over 2m50s
1454 runs so far, 0 failures, over 2m55s
1493 runs so far, 0 failures, over 3m0s
1534 runs so far, 0 failures, over 3m5s
1575 runs so far, 0 failures, over 3m10s
1614 runs so far, 0 failures, over 3m15s
1651 runs so far, 0 failures, over 3m20s
1696 runs so far, 0 failures, over 3m25s
1708 runs completed, 1 failures, over 3m26s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/gossip/gossip_test.go
+++ b/gossip/gossip_test.go
@@ -281,6 +281,7 @@ func TestGossipOrphanedStallDetection(t *testing.T) {
 
 	peerNodeID := peer.GetNodeID()
 	peerAddr := peer.GetNodeAddr()
+	peerAddrStr := peerAddr.String()
 
 	local.startClient(peerAddr, peerNodeID)
 
@@ -290,7 +291,16 @@ func TestGossipOrphanedStallDetection(t *testing.T) {
 				return nil
 			}
 		}
-		return errors.Errorf("%d not yet connected", peerNodeID)
+		return errors.Errorf("node %d not yet connected", peerNodeID)
+	})
+
+	util.SucceedsSoon(t, func() error {
+		for _, resolver := range local.GetResolvers() {
+			if resolver.Addr() == peerAddrStr {
+				return nil
+			}
+		}
+		return errors.Errorf("node %d descriptor not yet available", peerNodeID)
 	})
 
 	local.bootstrap()
@@ -301,7 +311,7 @@ func TestGossipOrphanedStallDetection(t *testing.T) {
 	util.SucceedsSoon(t, func() error {
 		for _, peerID := range local.Outgoing() {
 			if peerID == peerNodeID {
-				return errors.Errorf("%d still connected", peerNodeID)
+				return errors.Errorf("node %d still connected", peerNodeID)
 			}
 		}
 		return nil
@@ -317,6 +327,6 @@ func TestGossipOrphanedStallDetection(t *testing.T) {
 				return nil
 			}
 		}
-		return errors.Errorf("%d not yet connected", peerNodeID)
+		return errors.Errorf("node %d not yet connected", peerNodeID)
 	})
 }

